src/titanium-company-selector.ts

```diff
@@ -0,0 +1,228 @@
+import '@vaadin/vaadin-combo-box/theme/material/vaadin-combo-box-light';
+import '@vaadin/vaadin-text-field/theme/material/vaadin-text-field';
+import '@leavittsoftware/api-service/lib/api-service-element';
+
+import {ApiServiceElement} from '@leavittsoftware/api-service/lib/api-service-element';
+import {ODataDto} from '@leavittsoftware/api-service/lib/odata-dto';
+import {customElement, observe, property, query} from '@polymer/decorators';
+import {html, PolymerElement} from '@polymer/polymer';
+
+export interface Company extends ODataDto {
+  Id: number;
+  Roles: Array<CompanyRole>;
+  Names: Array<CompanyName>;
+}
+
+export interface CompanyRole extends ODataDto {
+  '@odata.type': string;
+  'Id': number;
+}
+
+export interface CompanyName extends ODataDto {
+  Name: string;
+}
+
+export type companyComboBoxItem = {
+  label: string,
+  value: Partial<Company>
+};
+
+@customElement('titanium-company-selector')
+export class TitaniumCompanySelectorElement extends PolymerElement {
+  @property() isLoading: boolean = false;
+
+  @property() controllerNamespace: string;
+
+  @property() opened: boolean;
+  @property() label: string|null;
+  @property() placeholder: string|null = 'Search...';
+  @property() companyId: number;
+
+  @property() disableAutoload: boolean = false;
+
+  @property() searchTerm: string;
+  @property() items: Array<companyComboBoxItem>;
+  @property({type: Object, notify: true})
+  selectedCompany: companyComboBoxItem|string = '';
+
+  @query('api-service') apiService: ApiServiceElement;
+
+  @observe('companyId', 'items')
+  async companyIdChanged(companyId: number|undefined) {
+    if (!companyId ||
+        (this.selectedCompany &&
+         (this.selectedCompany as companyComboBoxItem).value.Id ===
+             companyId)) {
+      return;
+    }
+
+    if (!this.isLoading && !this.items.length &&
+        this.disableAutoload !== false) {
+      this.items = await this._getCompanies();
+    } else {
+      return;
+    }
+
+    // restore selected company from company id
+    const companyItem = this.items.find(v => companyId === v.value.Id);
+    if (!companyItem) {
+      this.reportError(`No company with the Id ${companyId} could be found.`);
+    } else {
+      this.selectedCompany = companyItem;
+    }
+  }
+
+  private reportError(error: string) {
+    this.dispatchEvent(new CustomEvent(
+        'titanium-company-selector-error',
+        {bubbles: true, composed: true, detail: {message: error}}));
+  }
+
+  private async _getCompanies() {
+    this.isLoading = true;
+    let returnValue = new Array<companyComboBoxItem>();
+
+    try {
+      const result: Array<Company> =
+          (await this.apiService.getAsync<Company>(
+               `Companies?$expand=Names($filter=not IsExpired and CompanyNameType eq 'Main';$select=Name;$top=1),Roles($filter=not IsExpired and isof('LG.Core.DataModel.Core.AgencyRole');$select=Id;$top=1)&$select=Id&$filter=not IsExpired`,
```

Comment:
Need to be able to set filter, expand, etc. from the outside like person selector in marketing does

Reply:
The person selector in marketing does not...

Reply:
Sorry. Like the account-selector does. This person-selector definitely needs it because departments.leavitt.com will need to expand to employeeRoles and put a pretty specific filter on it. If these are going to be general use they probably both need to have the odata configurable. Also, we should remove the isof('LG.Core.DataModel.Core.AgencyRole') part of the company-selector expand.